shared/serdes/index.js

```diff
@@ -0,0 +1,155 @@
+const raw = Symbol('raw')
+export const serdesTagSymbol = Symbol('tag')
+export const serdesSerializeSymbol = Symbol('serialize')
+export const serdesDeserializeSymbol = Symbol('deserialize')
+
+// This file is at attempt at addressing the lack of support for
+// custom objects in `structuredClone`. See <https://github.com/whatwg/html/issues/7428>.
+// We need this so that certain custom objects can be shared using a
+// `MessagePort`. To use this functionality, the `serializer` function must be
+// called on the _sending_ side and the `deserializer` function must be called
+// on the _receiving_ side.
+// Note that it's paramount that the _receiving_ side call `deserializer.register`
+// with all of the possible types that are supported.
+// For how to implement `serialize` and `deserialize` support for custom types,
+// see the example below for a class `X`.
```

Comment:
Where is this example? I do not see it. It should probably be placed here near the top

Reply:
Added it back.